pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/defaultcolumn/BaseDefaultColumnHandler.java

```diff
@@ -188,13 +197,22 @@ public void updateDefaultColumns()
           defaultValueInSchema = fieldSpecInSchema.getDefaultNullValue().toString();
         }
 
-        if (dataTypeInMetadata != dataTypeInSchema || isSingleValueInMetadata != isSingleValueInSchema
-            || !defaultValueInSchema.equals(defaultValueInMetadata)) {
-          if (fieldTypeInMetadata == FieldSpec.FieldType.DIMENSION) {
-            defaultColumnActionMap.put(column, DefaultColumnAction.UPDATE_DIMENSION);
-          } else {
-            Preconditions.checkState(fieldTypeInMetadata == FieldSpec.FieldType.METRIC);
-            defaultColumnActionMap.put(column, DefaultColumnAction.UPDATE_METRIC);
+        if (fieldTypeInMetadata == FieldSpec.FieldType.DIMENSION) {
+          if (dataTypeInMetadata != dataTypeInSchema) {
+            defaultColumnActionMap.put(column, DefaultColumnAction.UPDATE_DIMENSION_DATA_TYPE);
```

Comment:
why allow incompatible schema operations?

Reply:
We aren't a

> > Default value indices for column: appHotLaunchDurationT cannot be removed for V3 format segment.
> 
> @siddharthteotia I see, I thought you're going to adjust this error message above, since update action will also print the message with word "removed", which is also misleading.

@jackjlli , thanks for catching this. I didn't realize that exception message will still print REMOVE. Made the necessary changes.

Reply:
We aren't allowing incompatible schema operations. This behavior doesn't change any behavior. It's just that exact nature of UPDATE operation is captured. 

I think you are suggesting to throw exception for remove and update here itself in BaseDefaultColumnHandler when we compute the default column action. I don't think that can be done since for V1 we sill allow update/remove